addons/mrp/models/mrp_workorder.py

```diff
@@ -195,9 +227,66 @@ def generate_wo_lines(self):
             line_values = self._generate_lines_values(move, qty_to_consume)
             self.workorder_line_ids |= self.env['mrp.workorder.line'].create(line_values)
 
-    def _assign_default_final_lot_id(self):
-        self.final_lot_id = self.env['stock.production.lot'].search([('use_next_on_work_order_id', '=', self.id)],
-                                                                    order='create_date, id', limit=1)
+    def _update_wo_lines(self):
+        """ update existing line on the workorder. It could be trigger manually
+        after a modification of qty_producing.
+        """
+        self.ensure_one()
+        line_values = self._update_workorder_lines()
+        self.workorder_line_ids |= self.workorder_line_ids.create(line_values['to_create'])
+        if line_values['to_delete']:
+            line_values['to_delete'].unlink()
+        for line, vals in line_values['to_update'].items():
+            line.write(vals)
+
+    def _action_assign(self):
```

Comment:
When reserving a manufacturing order after planning its work orders, as the work order lines are already created, the user won't be suggested the lots of the reservation. To update the suggestions, we chose to
- unlink the work order lines if nothing was done and then regenerate them according to the new reservation
- split the work order lines if some work was done (to keep what's done) and then regenerate work order lines for what's left to do according to the new reservation.

If the user reserves then plans, this code does nothing.

Reply:
rename to _refresh_wo_lines and make sure it sits next to _generate_wo_lines